scripts.py

```diff
@@ -2242,6 +2245,82 @@ def explain_work(cls, work):
             print "  %s: %r" % (active, pool.identifier)
 
 
+class FixInvisibleWorksScript(Script):
+    """Try to figure out why Works aren't showing up.
+
+    This is a common problem on a new installation.
+    """
+    def __init__(self, _db=None, output=None, search=None):
+        super(FixInvisibleWorksScript, self).__init__(_db)
+        self.output = output or sys.stdout
+        self.search = search or ExternalSearchIndex(_db)
+    
+    def run(self):
+        ready = self._db.query(Work).filter(Work.presentation_ready==True)
+        unready = self._db.query(Work).filter(Work.presentation_ready==False)
+        ready_count = ready.count()
+        unready_count = unready.count()
+        self.output.write("%d presentation-ready works.\n" % ready_count)
+        self.output.write("%d works not presentation-ready.\n" % unready_count)
+
+        if unready_count > 0:
+            self.output.write(
+                "Attempting to make %d works presentation-ready based on their metadata.\n" % (unready_count)
+            )
+            for work in unready:
+                work.set_presentation_ready_based_on_content(self.search)
+
+        ready_count = ready.count()
+
+        if unready_count > 0:
+            self.output.write(
+                "%d works are now presentation-ready.\n" % ready_count
```

Comment:
Is this to update that some formerly unready Works have become ready, but not all?